src/System.Net.Sockets.Legacy/tests/FunctionalTests/DualModeSocketTest.cs

```diff
@@ -1007,7 +1006,7 @@ private void Accept_Helper(IPAddress listenOn, IPAddress connectTo)
                 SocketClient client = new SocketClient(serverSocket, connectTo, port);
                 Socket clientSocket = serverSocket.Accept();
                 Assert.True(clientSocket.Connected);
-                Assert.True(clientSocket.DualMode);
+                Assert.True((listenOn != IPAddress.IPv6Any && !listenOn.IsIPv4MappedToIPv6) || clientSocket.DualMode);
```

Comment:
Or it could be done with RuntimeInformat.IsOSPlatform, e.g.
```C#
if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
{
    Assert.True(...);
}
else
{
    Assert.True(...);
}
```
Up to you guys.  I'm ok with whatever you decide.